snapcraft/plugins/make.py

```diff
@@ -83,12 +83,13 @@ def schema(cls):
             'default': [],
         }
 
+        return schema
+
+    @classmethod
+    def get_build_properties(cls):
         # Inform Snapcraft of the properties associated with building. If these
         # change in the YAML Snapcraft will consider the build step dirty.
-        schema['build-properties'].extend(
-            ['makefile', 'make-parameters', 'make-install-var'])
-
-        return schema
+        return ['makefile', 'make-parameters', 'make-install-var', 'artifacts']
```

Comment:
Note that, after #978, this will overwrite the `cmake` build properties. This PR is likely the place to fix that (`cmake` should be building on `make`'s version).